designer/server/lib/persistence/dynamoDBPersistenceService.ts

```diff
@@ -0,0 +1,190 @@
+import * as AWS from "aws-sdk";
+import config from "../../config";
+import { PersistenceService } from "./persistenceService";
+import { Logger, FormConfiguration } from "@xgovformbuilder/model";
+import { ConfigurationOptions } from "aws-sdk/lib/config-base";
+import { DocumentClient } from "aws-sdk/clients/dynamodb";
+
+const awsConfig: ConfigurationOptions = {
+  region: "eu-west-2",
+  secretAccessKey: config.awsSecretKey,
+  accessKeyId: config.awsAccessKeyId,
+};
+
+export class dynamoDBPersistenceService implements PersistenceService {
+  logger: any;
+  client: DocumentClient;
+
+  constructor(server: any) {
+    this.logger = new Logger(server, "dynamoDBPersistenceService");
+    AWS.config.update(awsConfig);
+    this.client = new AWS.DynamoDB.DocumentClient({ region: config.awsRegion });
+  }
+
+  async listAllConfigurations(): Promise<FormConfiguration[]> {
+    return new Promise(async (resolve, reject) => {
+      if (!config.dynamoDBTable) {
+        this.logger.error(
+          `No table name was supplied. To use the Dynamo DB service, you must supply the table name the forms will be built to.`
+        );
+      } else {
+        try {
+          await this.client
```

Comment:
I believe with the new aws sdk (v3) allows you to use async await if you want to ditch this https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/using-async-await.html